config/config.go

```diff
@@ -65,34 +70,95 @@ func ReadConfig(configFilePath string) (*Configuration, *common.Block, error) {
 			return nil, nil, errors.Wrapf(err, "failed to read shared config, path is empty")
 		}
 	case "block":
-		if conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File != "" {
-			blockPath := conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File
-			data, err := os.ReadFile(conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File)
+		// If node is router or batcher, check if config store has blocks, and if yes bootstrap from the last block
+		if nodeRole == "router" || nodeRole == "batcher" {
+			configStore, err = configstore.NewStore(conf.LocalConfig.NodeLocalConfig.FileStore.Path)
 			if err != nil {
-				return nil, nil, fmt.Errorf("could not read block %s", blockPath)
+				return nil, nil, fmt.Errorf("failed creating router config store: %s", err)
 			}
-			genesisBlock, err = protoutil.UnmarshalBlock(data)
+
+			listBlocks, err := configStore.ListBlocks()
```

Comment:
This is reading all the blocks, which is unnecessary, you can `ListBlockNumbers` and then `GetByNumber`.

Another option is to call "Last()" and catch the error `"config store is empty"`.